dot/parachain/bitfield-distribution/bitfield_distribution.go

```diff
@@ -324,13 +326,32 @@ func (b *BitfieldDistribution) processPeerViewChangeEvent(event networkbridgeeve
 	}
 }
 
-func (b *BitfieldDistribution) processOurViewChangeEvent(event networkbridgeevents.OurViewChange) error {
-	//TODO implement in #4359
-	panic("implement me")
+func (b *BitfieldDistribution) processOurViewChangeEvent(event networkbridgeevents.OurViewChange) {
+	logger.Tracef("process OurViewChange event: %v", event)
+
+	oldView := b.ourView
+	b.ourView = event.View
+
+	// in the new view but not in the old view
+	for _, added := range b.ourView.Difference(oldView) {
+		// newly added views will be handled in the active leaves update handler, ideally this should never happen
+		if b.perRelayParent[added] == nil {
+			logger.Errorf("our view contains %s, but not in active heads", added.String())
+		}
+	}
+
+	// in the old view but not in the new view
+	for _, removed := range oldView.Difference(b.ourView) {
+		b.mu.Lock()
+		delete(b.perRelayParent, removed)
+		b.mu.Unlock()
+	}
```

Comment:
```suggestion
        b.mu.Lock()
        defer b.mu.Unlock()
	for _, removed := range oldView.Difference(b.ourView) {
		delete(b.perRelayParent, removed)
	}
```

Reply:
lol good catch man, def shouldn't do the lock and ulock for each round in a loop, will change it